torch/utils/_benchmark/utils/timer.py

```diff
@@ -134,3 +137,36 @@ def stop_hook(times):
         times = self._threaded_measurement_loop(number, time_hook, stop_hook, min_run_time=min_run_time,
                                                 callback=callback)
         return self._construct_measurement(number_per_run=number, times=times)
+
+    def collect_callgrind(self, number=100, standardize=True):
+        if not isinstance(self._stmt, str):
+            raise ValueError("`collect_callgrind` currently only supports string `stmt`")
+
+        # __init__ adds torch, and Timer adds __builtins__
+        allowed_keys = {"torch", "__builtins__"}
+        if any(k not in allowed_keys for k in self._globals.keys()):
+            raise ValueError(
+                "`collect_callgrind` does not currently support passing globals. "
+                "Please define a `setup` str instead.")
+
+        if self._globals.get("torch", torch) is not torch:
+            raise ValueError("`collect_callgrind` does not support mocking out `torch`.")
+
+        # Check that the statement is valid. It doesn't guarantee success, but it's much
+        # simpler and quicker to raise an exception for a faulty `stmt` or `setup` in
+        # the parent process rather than the valgrind subprocess.
+        self._timer.timeit(1)
+        fn_counts = valgrind_wrapper.wrapper_singleton().collect_callgrind(
+            stmt=self._stmt, setup=self._setup, number=number, num_threads=self._num_threads)
+
+        if standardize:
+            standardized_fn_counts = []
+            prefix_truncations = ["build/aten/", "work/Python/", "work/Objects/"]
```

Comment:
where does `work/Python` come from? is it some path in the CPython build or is it specific to particular build of python you have?

Reply:
`Objects` and `Python` come from CPython. I dropped the `work` prefix as it might have been specific to my system, and added a comment about the CPython origins.